source/crud/compound-operations.txt

```diff
@@ -0,0 +1,52 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+Usually client applications only need to read data out of a database or
+write data into a database. However, client applications sometimes need
+to read and write data in a single interaction with the database. While
+you can accomplish this task using separate reads and writes, doing so
+requires the client application to gracefully handle potential errors
+at any stage of the process and in multiple potential error states.
+This increases the complexity of your code and can make your client
+application brittle and difficult to test.
+
+Compound operations combine read and write operations
+in a single atomic statement, so there's no chance of data changing in
+between a read and a subsequent write; in fact, both operations take
+place in the same line of code from the perspective of your client
+application.
+
+This property can be useful in cases where you want to write to a
+specific document, but you haven't found it yet. If you just perform a
+read for the document's ``_id`` and then try to alter the document you
+just found, it's possible that someone else can alter the document in
+between your read and write operations. This doesn't stop you from doing
+this work, but it can make error handling much more difficult. Compound
+operations help keep your logic straightforward by handling that logic
+entirely inside the database behind a layer of abstraction, so you don't
+have to worry about it.
```

Comment:
True... but should we really encourage our users to make use of transactions? I understand that there are a lot of valid use cases, but for your average introductory user transactions will just lead them astray when they could use these methods instead and avoid a lot of potential error cases.

I'd be fine with adding a section toward the bottom that talks about transactions, and how transactions essentially allow you to compose your own versions of these compound operations... but I'd probably want a use case outside of findOneAndUpdate/Replace/Delete to do that. Maybe we could talk about ``findMultipleAndDelete``? Would appreciate any suggestions.

Reply:
Going to break the "compound operations" page into Transactions vs. built-in methods as a starting point. I think "compound operations" actually encapsulates both, so this might be a good way to bring in transactions too without pushing users too far toward them